### PR TITLE
PIM-10167: Import file - datetime in price attribute break the import

### PR DESCRIPTION
**Pull request by BryanS29-akeneo**
**Description (for Contributor and Core Developer)**

Throw an exception when the value of price attribute is interpreted as DateTime in import.

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Component/Product/Connector/ArrayConverter/FlatToStandard/ColumnsMerger.php [hunk 1]
```diff
@@ -58,6 +60,10 @@ public function merge(array $row)
                 if (AttributeTypes::BACKEND_TYPE_METRIC === $attribute->getBackendType()) {
                     $collectedMetrics = $this->collectMetricData($collectedMetrics, $attributeInfos, $fieldValue);
                 } elseif (AttributeTypes::BACKEND_TYPE_PRICE === $attribute->getBackendType()) {
+                    if ($fieldValue instanceof \DateTimeInterface) {
+                        throw new BusinessArrayConversionException("Can not convert cell {$fieldName} with date format to attribute of type text", "pim_import_export.notification.export.warnings.xlsx_cell_date_to_text_conversion_error", [$fieldName]);
```
Comment (pchasle) [inline src/Akeneo/Pim/Enrichment/Component/Product/Connector/ArrayConverter/FlatToStandard/ColumnsMerger.php:64]: Don't you mean "attribute of type price"?

Comment (BryanS29-akeneo) [reply src/Akeneo/Pim/Enrichment/Component/Product/Connector/ArrayConverter/FlatToStandard/ColumnsMerger.php:64]: I thought like that : the attribute is price and when we fill a value inside it's of type text. But maybe I'm wrong, is price a specific type ?

Comment (nmarniesse) [reply src/Akeneo/Pim/Enrichment/Component/Product/Connector/ArrayConverter/FlatToStandard/ColumnsMerger.php:64]: I would use "attribute of type price" too

Comment (nmarniesse) [reply src/Akeneo/Pim/Enrichment/Component/Product/Connector/ArrayConverter/FlatToStandard/ColumnsMerger.php:64]: Can you add a test in ColumnsMergerSpec to check the `BusinessArrayConversionException` is thrown in this case?

**Review by nmarniesse — APPROVED** [review]
GTM after fix

**Review by RiffVentura — APPROVED** [review]

**Review by ghoussard — APPROVED** [review]

**Review by LaurentPetard — COMMENTED** [review]
No conflicts

**Pull request merged**